manage

```diff
@@ -687,7 +687,7 @@ test.pylint() {
 test.pyright() {
     build_msg TEST "[pyright] static type check of python sources"
     nodejs.ensure
-    pyenv.cmd pyright
+    pyright
```

Comment:
Why do you think `pyenv.cmd` is not needed here?

Reply:
Why do you think it is needed here? :P

Reply:
> Pyright isn't written in Python, it's written in Typescript.

This was what I needed to know, to give the right feedback: the python environment (virtualenv) is needed ...

```
{
    "venvPath": "local",
    "venv": "py3",
...
```

... even if it is a static type checker does not need the python interpreter itself.

with the pyenv.cmd we ensure that the virtualenv in `./local` is build.

By example first clean an than run that test target ..

    make clean test.pyright

In such cases there is no `./local/py3` and pyright alternatively uses the python environment from your HOME or if there is no special environment in your HOME it uses the python environment from the OS.

Reply:
Ah yeah that makes sense. I still think that it would be clearer to just do a `pyenv.install` before.

Reply:
I rebased the PR accordingly.

Reply:
> Ah yeah that makes sense. I still think that it would be clearer to just do a `pyenv.install` before.

Sadly this is not enough .. `pyenv.install`  will install virtualenv but not activate.  Don't mix python environment with the nvm environment where no activation is needed.

Reply:
The virtualenv doesn't need to be active for pyright ... it's a static code analyzer written in TypeScript.

Reply:
> The virtualenv doesn't need to be active for pyright ... it's a static code analyzer written in TypeScript, it does not actually execute any Python code.

Even if it is static, the analyser needs to import the modules from the environment to get the types.

Reply:
Yes but afaik Pyright discovers the modules itself via the `venvPath` & `venv` settings in `pyrightconfig.json`. Afaik it does not use (or need) environment variables to do so.

Reply:
> Afaik it does not use (or need) environment variables to do so.

"AFAIK" .. this is what I also know/thought .. but we don't know it .. and I think for those do not know the details, its hard to understand if we use the python environment for all py-processes but not for the analyzer. The cost is not high so I think it is more robust and predictable if we add the `pyenv.cmd` in the front.  This guarantees that all (import) actions take place in this environment.

Reply:
> but we don't know it

I am 99% sure.

Pyright has absolutely no dependencies on Python.

Reply:
>at least the python version is a dependence.

No it's not, see the [docs](https://github.com/microsoft/pyright/blob/main/docs/configuration.md):

>pythonVersion [string, optional]: Specifies the version of Python that will be used to execute the source code. The version should be specified as a string in the format "M.m" where M is the major version and m is the minor (e.g. "3.0" or "3.6"). If a version is provided, pyright will generate errors if the source code makes use of language features that are not supported in that version. It will also tailor its use of type stub files, which conditionalizes type definitions based on the version. If no version is specified, pyright will use the version of the current python interpreter, if one is present.

Reply:
> The version should be specified as a string in the format "M.m" 

Thanks for clarify!

> The question is how the current python interpreter is determined thinking

Thats why I thought it is better we integrate the CI test in the python tests:

https://github.com/searxng/searxng/blob/96a1f79c6dcaa08a231ad3a28b8c17b677cddc9c/.github/workflows/integration.yml#L16

Reply:
Ah ok ... we were both correct :P 

You were right that we should execute it in the env and I was right that pyright doesn't need to be run in the env. But we still should because it executes whatever `python` is on the PATH to determine the Python version.

I'll rebase the PR.

Reply:
> Ah ok ... we were both correct :P

:-)  :-) ...  

>  If no version is specified, pyright will use the version of the current python interpreter, if one is present.

When we run the tests in the different python version, I think we do not need to define a version for pyright.

> I'll rebase the PR.

Thanks! .. I'am looking forward :-)